core/server/api/authentication.js

```diff
@@ -216,8 +216,8 @@ authentication = {
         }
 
         function sendResetNotification(data) {
-            var baseUrl = config.get('forceAdminSSL') ? (config.get('urlSSL') || config.get('url')) : config.get('url'),
-                resetUrl = globalUtils.url.urlJoin(baseUrl, 'ghost/reset', globalUtils.encodeBase64URLsafe(data.resetToken), '/');
+            var baseUrl = config.get('forceAdminSSL') ? globalUtils.url.urlFor('home', {secure: true}, true) : globalUtils.url.urlFor('home', true),
+                resetUrl = globalUtils.url.urlJoin(baseUrl.replace(/\/$/, ''), '/ghost/reset/', globalUtils.encodeBase64URLsafe(data.resetToken), '/');
```

Comment:
If you are using `urlJoin`, you don't have to manually delete the trailing slash from the baseUrl.